op-batcher/batcher/driver.go

```diff
@@ -576,19 +597,10 @@ func (l *BatchSubmitter) throttlingLoop(ctx context.Context) {
 		}
 	}
 
-	cachedPendingBytes := int64(0)
-	for {
-		select {
-		case <-ticker.C:
-			updateParams(int64(cachedPendingBytes))
-		case pendingBytes := <-l.pendingBytesUpdated:
-			cachedPendingBytes = pendingBytes
-			updateParams(pendingBytes)
-		case <-ctx.Done():
-			l.Log.Info("DA throttling loop done")
-			return
-		}
+	for pendingBytes := range pendingBytesUpdated {
+		updateParams(pendingBytes)
 	}
```

Comment:
if we add a signal hook to the active endpoint provider (AEP), we could receive the signal here, and then reuse the latest pendingBytes value that we locally store in a variable, like
```go
for {
		select {
		case <-activeChanged:
			updateParams(int64(cachedPendingBytes))
		case pendingBytes, ok := <-l.pendingBytesUpdated:
			if !ok {return;} // or break
			cachedPendingBytes = pendingBytes
			updateParams(pendingBytes)
}
```
where `activeChanged` would be the channel on wich the AEP signals. So we'd basically replace the ticker by the AEP signal and shut down the routine by a signal that the pending bytes update channel got closed.

Reply:
Yeah I like this. Will need to put back the `cachedPendingBytes` var, but it makes sense to keep it so we can prod the active sequencer with up-to-date information.

Reply:
[2988863](https://github.com/ethereum-optimism/optimism/pull/14244/commits/2988863da57b37f4d3cfbef36acbf11037767a0e)